src/Files.App/Views/LayoutModes/ColumnViewBrowser.xaml.cs

```diff
@@ -435,20 +435,33 @@ private void ColumnViewBase_ItemTapped(object? sender, EventArgs e)
 			var column = sender as ColumnParam;
 			if (column?.ListView.FindAscendant<ColumnViewBrowser>() != this || string.IsNullOrEmpty(column.NavPathParam))
 				return;
-			
+
 			CloseUnnecessaryColumns(column);
 		}
 
 		private void CloseUnnecessaryColumns(ColumnParam column)
 		{
-			var columnPath = ((ColumnHost.ActiveBlades.Last().Content as Frame)?.Content as ColumnShellPage)?.FilesystemViewModel?.WorkingDirectory;
-			var columnFirstPath = ((ColumnHost.ActiveBlades.First().Content as Frame)?.Content as ColumnShellPage)?.FilesystemViewModel.WorkingDirectory;
-			if (string.IsNullOrEmpty(columnPath) || string.IsNullOrEmpty(columnFirstPath))
+			if (string.IsNullOrEmpty(column.NavPathParam))
 				return;
 
-			var destComponents = StorageFileExtensions.GetDirectoryPathComponents(column.NavPathParam);
-			var (_, relativeIndex) = GetLastCommonAndRelativeIndex(destComponents, columnPath, columnFirstPath);
-			if (relativeIndex >= 0)
+			var relativeIndex = column.Column is not 0 ? column.Column : -1;
+
+			if (column.Source is not null)
+			{
+				for (var i = 0; i < ColumnHost.ActiveBlades.Count && relativeIndex is -1; i++)
+				{
+					var bladeColumn = ColumnHost.ActiveBlades[i].FindDescendant<ColumnViewBase>();
+					if (bladeColumn is not null && bladeColumn == column.Source)
+						relativeIndex = i;
+				}
+			}
+
+			if (relativeIndex is -1)
+				// Get the index of the blade with the same path as the requested
+				while (relativeIndex < ColumnHost.ActiveBlades.Count &&
+					!column.NavPathParam.Equals(GetWorkingDirOfColumnAt(++relativeIndex))) ;
+
+			if (relativeIndex >= 0 && relativeIndex < ColumnHost.ActiveBlades.Count)
```

Comment:
```suggestion
			if (relativeIndex is -1)
			{
				// Get the index of the blade with the same path as the requested
				var bladeColumn = ColumnHost.ActiveBlades.FirstOrDefault(x
					=> column.NavPathParam.Equals(((x.Content as Frame)?.Content as ColumnShellPage)?.FilesystemViewModel?.WorkingDirectory));

				if (bladeColumn is not null)
					relativeIndex = ColumnHost.ActiveBlades.IndexOf(bladeColumn);
			}

			if (relativeIndex >= 0)
```
For example.

Reply:
At first, I used a while because I thought that scanning through the blades twice was a waste of time, but as you pointed out that was not that readable, and so I'll use your code. Thank you